hatch/src/cli/commands/build.rs

```diff
@@ -1,26 +1,26 @@
-use hatch_error::{ HatchResult };
-use clap::{ ArgMatches };
+use clap::ArgMatches;
 use cli::commands::Command;
 use generators::Generator;
-use generators::tup::Tup;
+use hatch_error::HatchResult;
+use generators::tup::make_a_tup_in_a_box;
 
 pub struct Build;
 
 impl<'build> Build {
-  pub fn new() -> Build {
-    Build
-  }
+    pub fn new() -> Build {
+        Build
+    }
 }
 
 impl<'command> Command<'command> for Build {
-  fn execute(&self, args: &ArgMatches<'command>) -> HatchResult<()> {
-    let (project_path, project) = self.read_project_context(args)?;
-    println!("Building project...\n");
-    self.build(&project_path)?;
-    println!("Generating assets...\n");
-    // TODO: make a trait object so we can dynamic dispatch on a command line arg for generator
-    let generator: Box<Generator> = Box::new(Tup{});
-    self.generate_assets(generator, project_path, &project)?;
-    Ok(())
-  }
+    fn execute(&self, args: &ArgMatches<'command>) -> HatchResult<()> {
+        let (project_path, project) = self.read_project_context(args)?;
+        println!("Building project...\n");
+        self.build(&project_path)?;
+        println!("Generating assets...\n");
+        // TODO: make a trait object so we can dynamic dispatch on a command line arg for generator
+        let generator = make_a_tup_in_a_box();
```

Comment:
I added this factory function to help with easy refactoring.

If you don't like the name, we can change it.